jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/Label.java

```diff
@@ -20,12 +20,18 @@
 import jsettlers.graphics.utils.UIPanel;
 
 public class Label extends UIPanel {
+    public enum HorizontalAlignment {
+        LEFT,
+        CENTRE,
+        RIGHT,
+    }
 
 	private final EFontSize size;
-	private final String[] words;
+	private String[] words;
```

Comment:
Front ends are typically single threaded to keep it responsive and avoid lock ups. I think adding synchronisation blocks might encourage the wrong type of architecture.